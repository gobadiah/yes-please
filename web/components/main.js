import React from 'react';
import {
  oneOfType,
  element,
  arrayOf,
} from 'prop-types';

// eslint-disable-next-line no-unused-vars
import * as styles from '../styles';

class Main extends React.Component {
  render() {
    const { children } = this.props;

    return (
      <div>
        {children}
      </div>
    );
  }
}

Main.propTypes = {
  children: oneOfType([
    element,
    arrayOf(element),
  ]).isRequired,
};

export default Main;
