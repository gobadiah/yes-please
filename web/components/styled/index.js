import styled from 'react-emotion';

export const Flex = styled('div')`
  display: flex;
`;

export const FlexColumn = styled(Flex)`
  flex-direction: column;
`;

export const FlexColumnCenter = styled(FlexColumn)`
  align-items: center;
  justify-content: center;
`;
