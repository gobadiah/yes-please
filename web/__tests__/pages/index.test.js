import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Index from '../../pages';

import Main from '../../components/main';
import { FlexColumnCenter } from '../../components/styled';

describe('Index page', () => {
  it('should have a Main component as root', () => {
    const index = shallow(<Index />);
    expect(index.children).toHaveLength(1);
    expect(index.type()).toBe(Main);
    expect(index.childAt(0).type()).toBe(FlexColumnCenter);
  });

  it('should match snapshot', () => {
    const index = renderer.create(<Index />);
    const tree = index.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
