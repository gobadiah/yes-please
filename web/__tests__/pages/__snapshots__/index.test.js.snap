// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <div
    className="emotion-0 emotion-1"
  >
    Hello world
  </div>
</div>
`;
